Makefile

```diff
@@ -13,6 +13,7 @@ SOEXT ?= $(shell ruby -e 'puts RbConfig::CONFIG["SOEXT"]')
 CPPFLAGS := -Iinclude $(CPPFLAGS)
 CFLAGS := -g -O2 -std=c99 -Wall -Werror -Wextra -Wpedantic -Wundef -Wconversion -Wno-missing-braces -fPIC -fvisibility=hidden $(CFLAGS)
 CC ?= cc
+AR ?= ar
```

Comment:
As discovered in https://github.com/ruby/prism/pull/2711#discussion_r1569712785, `CC ?= cc` and `AR ?= ar` has no effect, whether run from `make` directly (because GNU `make` already preconfigures them to `cc` and `ar` respectively) or from `extconf.rb` (as it’ll override them with `rbconfig` values)
```suggestion
```

Reply:
I agree with you in theory, but something else must be going on here, because after lots of back and forth with @forthrin this is what finally resolved the issue. I'm assuming there's something else here going on.